src/Transport/Topology/TopologyCreator.cs

```diff
@@ -19,10 +19,11 @@ public TopologyCreator(ICreateAzureServiceBusSubscriptionsInternal subscriptions
             this.queuesCreator = queuesCreator;
             this.subscriptionsCreator = subscriptionsCreator;
             this.namespaces = namespaces;
+            var namespaceConfigurations = settings.Get<NamespaceConfigurations>(WellKnownConfigurationKeys.Topology.Addressing.Namespaces);
```

Comment:
Are we certain this constructor is called after settings are complete?

Reply:
Same answer as [here](https://github.com/Particular/NServiceBus.AzureServiceBus/pull/608#discussion_r135875073) will apply. Configuration object is finalized **prior** to endpoint running. Meaning message pump (which kicks off topology operator) will not be invoked before user-provided configuration  is captured.